lxd/daemon.go

```diff
@@ -334,12 +334,30 @@ func (d *Daemon) Authenticate(w http.ResponseWriter, r *http.Request) (trusted b
 	}
 
 	if d.oidcVerifier != nil && d.oidcVerifier.IsRequest(r) {
-		userName, err := d.oidcVerifier.Auth(d.shutdownCtx, w, r)
+		result, err := d.oidcVerifier.Auth(d.shutdownCtx, w, r)
 		if err != nil {
 			return false, "", "", err
```

Comment:
could we be wrapping the error here with more context, "Failed OIDC authentication: %w" or something like that?

Reply:
I've made this change in a separate commit. There was a type assertion on this error for `err.(*oidc.AuthError)`. I've now made the `oidc.Verifier` return the error by value and am using `errors.As` to inspect it.